objection_engine/v4/ace_attorney_scene.py

```diff
@@ -0,0 +1,1551 @@
+        )
+
+        if "on_characters_cast" in self.callbacks:
+            self.callbacks["on_characters_cast"](users_to_characters)
+
+        # Get music
+        if music_code in self.music_data:
+            music_pack = self.music_data[music_code]
+        else:
+            raise KeyError(
+                f'Music code "{music_code}" not found. Ensure that a folder for it exists in the "music" folder, and that it has a "config.toml" file.'
+            )
+
+        self.pages: list[DialoguePage] = []
+
+        # Start relaxed music
+        self.relaxed_track = join(music_code, choice(music_pack["relaxed"]))
+        self.tense_track = join(music_code, choice(music_pack["tense"]))
+        self.pages.append(
+            DialoguePage([DialogueAction(f"music start {self.relaxed_track}", 0)])
+        )
+
+        self.has_done_objection = False
+
+        # Add boxes for the dialogue
+        for i, comment in enumerate(comments):
+            self.pages.extend(
+                self.get_boxes_with_pauses(
+                    user_name=comment.user_name,
+                    character=users_to_characters[comment.effective_user_id],
+                    text=comment.text_content,
+                    evidence_path=comment.evidence_path,
+                )
+            )
+
+            if "on_comment_processed" in self.callbacks:
+                self.callbacks["on_comment_processed"](i, len(comments), comment)
+
+    def update_pose_for_sentence(self, sentence: Sentence, sprites: list[str]):
+        sentiment: dict = self.get_sentiment(sentence.raw)[0]
+        try:
+            sprite_cat = sentiment.get("label", "neutral")
+        except IndexError:
+            sprite_cat = "neutral"
+
+        self.current_character_animation = choice(
+            [
+                s
+                for s in sprites[sprite_cat]
+                if not self.avoid_spoiler_sprites
+                or "spoiler" not in sprites
+                or (self.avoid_spoiler_sprites and s not in sprites["spoiler"])
+            ]
+        )
+
+    def get_boxes_with_pauses(
+        self, user_name: str, character: str, text: str, evidence_path: str = None
+    ):
+        self.current_character_name = character
+        this_char_data = self.character_data["characters"][character]
+        location = this_char_data["location"]
+        gender = this_char_data["gender"]
+        sprites = this_char_data["sprites"]
+
+        # Split text into sentences
+        pg_text = Text(text)
+        sentences: list[Sentence] = pg_text.sentences
+
+        # Determine if this should have an objection
+        text_polarity_data = self.get_sentiment(pg_text.raw)[0]
+        polarity_type = text_polarity_data["label"]
+        polarity_confidence = text_polarity_data["score"]
+
+        do_objection = (
+            polarity_type == "negative"
+            and polarity_confidence > 0.5
+            and not self.has_done_objection
+        )
```

Comment:
Hi!

This behavior differs from the original one. In the original one there is a chance to fire an objection both in positive and negative comments (not so in neutral ones) although it's less usual to do it in positive it's still possible to.

Is there any reason you've changed it?
Thanks!